src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledPackageFeed.cs

```diff
@@ -39,20 +39,31 @@ public class InstalledPackageFeed : PlainPackageFeedBase
 
         public override async Task<SearchResult<IPackageSearchMetadata>> ContinueSearchAsync(ContinuationToken continuationToken, CancellationToken cancellationToken)
         {
-            var searchToken = continuationToken as FeedSearchContinuationToken;
-            if (searchToken == null)
-            {
-                throw new InvalidOperationException(Strings.Exception_InvalidContinuationToken);
-            }
+            var searchToken = ThrowIfNotFeedSearchContinuationToken(continuationToken);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            PackageIdentity[] feedItems = _installedPackages.GetLatest();
+            IPackageSearchMetadata[] searchItems = await DoSearchAsync(feedItems, searchToken, cancellationToken);
+
+            return CreateResult(searchItems);
+        }
+
+        internal async Task<IPackageSearchMetadata[]> DoSearchAsync<T>(IEnumerable<T> feedItems, FeedSearchContinuationToken searchToken, CancellationToken cancellationToken) where T : PackageIdentity
+        {
+            cancellationToken.ThrowIfCancellationRequested();
 
-            PackageIdentity[] packages = PerformLookup(_installedPackages.GetLatest(), searchToken);
+            T[] packages = PerformLookup(feedItems, searchToken);
 
             IEnumerable<IPackageSearchMetadata> items = await TaskCombinators.ThrottledAsync(
                 packages,
                 (p, t) => GetPackageMetadataAsync(p, searchToken.SearchFilter.IncludePrerelease, t),
                 cancellationToken);
 
-            return CreateResult(items);
+            //  The packages were originally sorted which is important because we Skip based on that sort
+            //  however the asynchronous execution has randomly reordered the set. So we need to resort. 
```

Comment:
```suggestion
            // The packages were originally sorted which is important because we skip based on that sort
            // however, the asynchronous execution has randomly reordered the set. So, we need to resort.
```